packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -203,7 +197,7 @@ function newClick(
   const eventCountsSubscription = trackEventCounts(lifeCycle)
   let state: ClickState = { status: ClickStatus.ONGOING }
   const frustrations = new Set<FrustrationType>()
-  let onStopCallback = noop
+  const onStopCallbacks: Array<() => void> = []
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Using an observable could simplify it.